src/core/net/srp_client.hpp

```diff
@@ -410,6 +410,27 @@ class Client : public InstanceLocator, private NonCopyable
      */
     void SetCallback(Callback aCallback, void *aContext);
 
+    /**
+     * This method gets the TTL used in SRP update requests.
+     *
+     * Note that this is the TTL that would be requested by the SRP client. Server may choose to accept a different TTL.
+     *
+     * @returns The TTL (in seconds).
+     *
+     */
+    uint32_t GetTtl(void) const { return mTtl ? mTtl : mLeaseInterval; }
+
+    /**
+     * This method sets the TTL used in SRP update requests.
+     *
+     * Changing the TTL does not impact the TTL of already registered services/host-info.
+     * It only changes any future SRP update messages (i.e adding new services and/or refreshes of existing services).
+     *
+     * @param[in] aTtl  The TTL (in seconds). If zero, the TTL is set to the lease interval.
+     *
+     */
+    void SetTtl(uint32_t aTtl) { mTtl = aTtl; }
```

Comment:
What's the relationship between TTL and Lease Time?

Should the TTL always be smaller than Lease Time?

Reply:
The lease time controls when the SRP server deletes an outdated SRP registration, while TTL remains the same meaning as in normal DNS records. The SRP spec may help: https://datatracker.ietf.org/doc/html/draft-ietf-dnssd-srp-13#section-3

Reply:
Current draft spec states that TTL should never be longer than the lease time, but not a must:

```
   TTLs sent in SRP Updates are advisory: they indicate the SRP client's
   guess as to what a good TTL would be.  SRP servers may override these
   TTLs.  SRP servers SHOULD ensure that TTLs are reasonable: neither
   too long nor too short.  The TTL should never be longer than the
   lease time (Section 4.1).  Shorter TTLs will result in more frequent
   data refreshes; this increases latency on the DNS-SD client side,
   increases load on any caching resolvers and on the authoritative
   server, and also increases network load, which may be an issue for
   constrained networks.  Longer TTLs will increase the likelihood that
   data in caches will be stale.  TTL minimums and maximums SHOULD be
   configurable by the operator of the SRP server.
```

Reply:
Should we stick to `SHOULD` requirement if there is no strong reason not to?

Reply:
The SHOULD in the text above applies to servers. The client values are advisory. The server is the one that technically assigns the value that is used.

Reply:
Regardless, I updated `GetTtl()` and relevant docs to indicate that lease interval is used if TTL is configured to a larger value.